packages/apollo-language-server/src/utilities/source.ts

```diff
@@ -181,3 +185,39 @@ export function getASTNodeAndTypeInfoAtPosition(
     return null;
   }
 }
+
+export function isFieldResolvedLocally(
+  fieldNode: FieldNode,
+  root: ASTNode
+): boolean | null {
+  const { loc } = fieldNode;
+  if (!loc) return null;
+
+  let underClientDirective: boolean = false;
+
+  function visitor(
+    node: FieldNode | InlineFragmentNode | FragmentDefinitionNode
+  ) {
+    if (
+      loc &&
+      node.loc &&
+      node.loc.start <= loc.start &&
+      loc.end <= node.loc.end &&
+      node.directives &&
+      node.directives.find(directive => directive.name.value === "client") !=
```

Comment:
When reducing arrays to true or false based on the contents:
i.e. do I have **at least one directive in here** that's named client?
i.e. is **every item in this array** an x?

We have the lesser-known, more semantic `Array.some()` and `Array.every()` functions that I'm a big fan of.

Reply:
Great point, I think `Array.some()` makes a lot of sense here